File: WebApplication/src/rdpzsd/dtos/parts/person-student-sticker/sticker-error.dto.ts
import { CourseType } from "src/rdpzsd/enums/parts/course.enum";
import { Semester } from "src/shared/enums/semester.enum";

export class StickerErrorDto {
    hasError = false;

    missingStudentSemesters: MissingStudentSemesterDto[] = [];
    otherErrors: CustomStickerErrorDto[] = [];
}

export class CustomStickerErrorDto {
    error: string;
    errorAlt: string;
}

export class MissingStudentSemesterDto {
    course: CourseType;
    studentSemester: Semester;
}

File: Rdpzsd.Models/Dtos/Rdpzsd/Parts/PersonStudentSticker/StickerErrorDto.cs
using Rdpzsd.Models.Enums;
using Rdpzsd.Models.Enums.Rdpzsd.Parts;
using System.Collections.Generic;

namespace Rdpzsd.Models.Dtos.Rdpzsd.Parts.PersonStudentSticker
{
    public class StickerErrorDto
    {
        public bool HasError { get; set; } = false;

        public List<MissingStudentSemesterDto> MissingStudentSemesters { get; set; } = new List<MissingStudentSemesterDto>();
        public List<CustomStickerErrorDto> OtherErrors { get; set; } = new List<CustomStickerErrorDto>();
    }

    public class CustomStickerErrorDto
    {
        public string Error { get; set; }
        public string ErrorAlt { get; set; }
    }

    public class MissingStudentSemesterDto
    {
        public CourseType Course { get; set; }
        public Semester StudentSemester { get; set; }
    }
}
